mindone/models/lora.py

```diff
@@ -0,0 +1,342 @@
+import logging
+from typing import List
+
+import mindspore as ms
+import mindspore.common.initializer as init
+from mindspore import nn
+from mindspore.nn.cell import Cell
+from mindspore.nn.layer.activation import get_activation
+from mindspore.ops.primitive import Primitive
+
+from ..utils.config import get_obj_from_str
+from ..utils.version_control import is_old_ms_version
+
+__all__ = [
+    "LoRADenseLayer",
+    "inject_trainable_lora",
+    "make_only_lora_params_trainable",
+    "merge_lora_to_model_weights",
+    "get_lora_params",
+]
```

Comment:
In this case, I believe it would be better to apply OOP principles to create a simpler and more understandable API. For example:
```python
model = SomeModel(**params)
lora_model = LoRAModel(model, **params)
lora_model.train_lora_only()
lora_model.merge_with_model()
...
```

Reply:
It can incur parameter name changes if wrap the model in to LoRAModel.

Reply:
I think it is fine to wrap the model. if `LoRAModel` is not a cell, and it is just a wrapper, then there is no name changing.

Reply:
It is feasible indeed. so the question is whether it is beneficial. In training, we use `inject_trainable_lora`, and in inference, we use `merge_lora_to_model_weights`. They are run in different launches and don't share common attributes. It's a little bit over-encapsulation to put them in class.